packages/legacy/src/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -178,20 +194,29 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
   const providerType = providerTypeField.value;
   const formValues = providerType ? forms[providerType].values : null;
   const isFormValid = providerType ? forms[providerType].isValid : false;
+  const isOpenStackFormValid = forms["openstack"].isValid && (forms["openstack"].fields.insecure.value || forms["openstack"].fields.caCertIfSecure.value);
   const isFormDirty = providerType ? forms[providerType].isDirty : false;
 
   // Combines fields of all 3 forms into one type with all properties as optional.
   // This way, we can conditionally show fields based on whether they are defined in form state
   // instead of duplicating the logic of which providers have which fields.
   const fields = providerType
     ? (forms[providerType].fields as Partial<
-        typeof forms.vsphere.fields & typeof forms.ovirt.fields & typeof forms.openshift.fields
+        typeof forms.vsphere.fields & typeof forms.ovirt.fields & typeof forms.openstack.fields & typeof forms.openshift.fields
       >)
     : null;
 
   const usernamePlaceholder = {
     vsphere: 'Example, administrator@vsphere.local',
     ovirt: 'Example, admin@internal',
+    openstack: 'Example, admin',
+    openshift: undefined,
+  }[providerType || ''];
+
+  const domainNamePlaceholder = {
+    vsphere: undefined,
+    ovirt: undefined,
+    openstack: 'Example, redhat.com',
```

Comment:
```suggestion
    openstack: 'Example, acme.com',
```

Reply:
actually it should be more in the line of:
`https://openstck-server.com:5000/v3`

Reply:
```suggestion
    openstack: 'Example, [redhat.com](http://controller:5000/v3)',
```

copied from:
https://docs.openstack.org/keystone/latest/install/keystone-install-obs.html#install-and-configure-components

note that outside the trusted network (openshift will be outside) this is not a good example, it will be `https` with differernt port and path

Reply:
+1
The `domainNamePlaceholder` was removed since there is no real need of it.
The openStack **hostname** field was replaced with a **URL** field with a  `urlPlaceholder` example of: `http://controller:5000/v3'`